src/hypothesis/strategies.py

```diff
@@ -334,17 +334,35 @@ def floats(
                     allow_nan
                 ))
 
-    min_value = try_convert(float, min_value, 'min_value')
-    max_value = try_convert(float, max_value, 'max_value')
-
     check_valid_bound(min_value, 'min_value')
     check_valid_bound(max_value, 'max_value')
+
+    min_arg, max_arg = min_value, max_value
```

Comment:
All this rigmarole does seem correct (and necessary), but I really wish there were a built in function for "cast to float upward/downward"!